scipy/stats/_sensitivity_analysis.py

```diff
@@ -0,0 +1,418 @@
+    the numerator of the Sobol' indices. Looking at :math:`x_3`, the variance
+    of the mean is zero leading to :math:`S_{x_3} = 0`. But we can further
+    observe that the variance of the output is not constant along the parameter
+    values of :math:`x_3`. This heteroscedasticity is explained by higher order
+    interactions. Moreover, an heteroscedasticity is also noticeable on
+    :math:`x_1` leading to an interaction between :math:`x_3` and :math:`x_1`.
+    On :math:`x_2`, the variance seems to be constant and thus null interaction
+    with this parameter can be supposed. This case is fairly simple to analyse
+    visually---although it is only a qualitative analysis. Nevertheless, when
+    the number of input parameters increases such analysis becomes unrealistic
+    as it would be difficult to conclude on high-order terms.
+
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> from scipy.stats import f_ishigami, qmc
+    >>> n_dim = 3
+    >>> p_labels = ['$x_1$', '$x_2$', '$x_3$']
+    >>> sample = qmc.Sobol(d=n_dim).random(1024)
+    >>> sample = qmc.scale(
+    ...     sample=sample,
+    ...     l_bounds=[-np.pi, -np.pi, -np.pi],
+    ...     u_bounds=[np.pi, np.pi, np.pi]
+    ... )
+    >>> output = f_ishigami(sample)
+    >>> mini = np.min(output)
+    >>> maxi = np.max(output)
+    >>> n_bins = 10
+    >>> bins = np.linspace(-np.pi, np.pi, num=n_bins, endpoint=False)
+    >>> dx = bins[1] - bins[0]
+    >>> fig, ax = plt.subplots(1, n_dim)
+    >>> for i in range(n_dim):
+    >>>     xi = sample[:, i]
+    >>>     ax[i].scatter(xi, output, marker='+')
+    >>>     ax[i].set_xlabel(p_labels[i])
+    >>>     for bin_ in bins:
+    >>>         idx = np.where((bin_ <= xi) & (xi <= bin_ + dx))
+    >>>         xi_ = xi[idx]
+    >>>         y_ = output[idx]
+    >>>         ave_y_ = np.mean(y_)
+    >>>         ax[i].plot([bin_ + dx / 2] * 2, [mini, maxi], c='k')
+    >>>         ax[i].scatter(bin_ + dx / 2, ave_y_, c='r')
+    >>> ax[0].set_ylabel('Y')
+    >>> plt.tight_layout()
+    >>> plt.show()
+
+    """
+    random_state = check_random_state(random_state)
+    A, B = sample_A_B(n=n, dists=dists, random_state=random_state)
+    AB = sample_AB(A=A, B=B)
+
+    f_A = func(A)
+    f_B = func(B)
+    f_AB = func(AB)
+
+    # Y = (Y - Y.mean()) / Y.std()
+
+    first_order, total_order = saltelli_2010(f_A, f_B, f_AB)
+
+    def saltelli_2010_(idx):
+        f_A_ = f_A[idx]
+        f_B_ = f_B[idx]
+
+        n, d = A.shape
+        f_AB_ = np.empty((d, *f_A_.shape))
+        for i in range(d):
+            f_AB_[i] = f_AB[np.array(idx)+i*n]
+        f_AB_ = f_AB_.reshape(-1, f_A_.shape[1])
+
+        # A_ = A[idx]
+        # B_ = B[idx]
+        # AB_ = sample_AB(A=A_, B=B_)
+        # f_AB_ = func(AB_)
+
+        return saltelli_2010(f_A_, f_B_, f_AB_)
+
+    bootstrap_result = bootstrap(
+        [np.arange(n)], saltelli_2010_, method="BCa",
+        batch=int(n*0.7), n_resamples=99
```

Comment:
Where does 0.7 come from?
Also, note that `batch` is the number of _resamples_ that are processed at the same time, each resample being of size `n`. Just wanted to make sure that was clear.

Reply:
I still think there's a misunderstand of `batch`.

Suppose we are taking the mean along the $m$ rows of a $m \times n$ array. You can vectorize the code to do all the rows simultaneously, `batch=m`. Or you could do 10 rows at a time, `batch=10`. Or a single row at a time, `batch=1`. There's not much need to split up the operation for taking the mean because the whole array is probably already in memory, but when we're taking resamples we generate `batch` new samples, each of the same size as the original sample. So we don't always want to do all of the resamples simultaneously because that could take too much memory.

I can see `batch` being _inversely_ proportional to the number `n` of observations per sample (i.e. to keep memory usage constant regardless of sample size). If the `batch` size is linearly proportional to sample size, then memory usage grows quadratically with sample size.

Reply:
ah ok then I should just remove the value here and let the default do it's work. I thought this was the number of samples for each resample.